comictaggerlib/cli.py

```diff
@@ -28,25 +28,26 @@
 from comicapi.comicarchive import ComicArchive, MetaDataStyle
 from comicapi.genericmetadata import GenericMetadata
 from comictaggerlib.cbltransformer import CBLTransformer
-from comictaggerlib.comicvinetalker import ComicVineTalker, ComicVineTalkerException
 from comictaggerlib.filerenamer import FileRenamer
 from comictaggerlib.issueidentifier import IssueIdentifier
 from comictaggerlib.resulttypes import IssueResult, MultipleMatch, OnlineMatchResults
 from comictaggerlib.settings import ComicTaggerSettings
 
+from sourcesapi.sourcesapi import SourcesAPI
+from sourcesapi.sourcebaseclass import SourceInfoError
+
 logger = logging.getLogger(__name__)
 
 
 def actual_issue_data_fetch(
-    match: IssueResult, settings: ComicTaggerSettings, opts: argparse.Namespace
+    match: IssueResult, settings: ComicTaggerSettings, opts: argparse.Namespace, sources_api: SourcesAPI
 ) -> GenericMetadata:
     # now get the particular issue data
     try:
-        comic_vine = ComicVineTalker()
-        comic_vine.wait_for_rate_limit = opts.wait_on_cv_rate_limit
-        cv_md = comic_vine.fetch_issue_data(match["volume_id"], match["issue_number"], settings)
-    except ComicVineTalkerException:
-        logger.exception("Network error while getting issue details.  Save aborted")
+        cv_md = sources_api.fetch_issue_data(match["volume_id"], match["issue_number"])
+    except SourceInfoError as e:
+        logger.exception("Issue with " + e.source + " while retrieving issue details. Save aborted.\n" +
+                         e.code_name + " Error: " + str(e.code) + "." + str(e.sub_code) + " - " + e.desc)
```

Comment:
This needs to be an fstring and SourceInfoError needs to  have __str__ and/or __repr__ defined
```suggestion
        logger.exception("Error retrieving issue details. Save aborted.\n{e}")
```
Please make similar changes to any other strings.

Reply:
I'm fine with the f string but I'm trying to understand the rest, sorry. Do you want the error to only return a single string with everything in instead of a dict with code, sub_code, desc? So, e would be:
```
Comic Vine error 1.2 This error happened
```
Instead of breaking each part down? And to return the single line would be the reason to override the standard __str__ Exception?

Reply:
__str__ and __repr__ are designed to be overridden take OSError for example
```pycon
>>> try:
...   f = open("/test")
... except OSError as e:
...   err = e
... 
>>> err
FileNotFoundError(2, 'No such file or directory')
>>> err.__repr__()
"FileNotFoundError(2, 'No such file or directory')"
>>> err.  # I tab completed here
err.args                err.characters_written  err.errno               err.filename            err.filename2           err.strerror            err.with_traceback(     
>>> str(err)
"[Errno 2] No such file or directory: '/test'"
>>> err.__str__()
"[Errno 2] No such file or directory: '/test'"
```
vs a base exception
```pycon
>>> er = Exception("test")
>>> er
Exception('test')
>>> er.
er.args             er.with_traceback(  
>>> er.__str__()
'test'
```
For exceptions it is standard to provide a string representation that is formatted appropriately for the exception as the __str__method
If you are not familiar here is a decent explanation of the __str__ and __repr__ methods
https://pythonprogramming.net/__str__-__repr__-intermediate-python-tutorial/

Reply:
With the new commit, should I move the name, code, etc. out to an args or is it okay as is?

Reply:
It looks great!